src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -279,7 +279,19 @@ public void addFunctionExpressionGraph(ExpressionGraph subGraph,
     exprVertexMap.put(subGraphInput, inputVertex);
 
     // Connect the function output vertex to the main graph
-    ExpressionEdge outputEdge = new ExpressionEdge(exprVertexMap.get(subGraphOutput), mainGraphOutput.getTarget());
+    ExpressionVertex outputVertex = exprVertexMap.get(subGraphOutput);
+    if (((TupleValueVertex) subGraphOutput).getValueEdges().size() == 1) {
```

Comment:
Why we are able to do this cast is not obvious to me. Is there a simple explanation that can be added in a comment, or can the type be more restricted at the function definition?